File: auxiliary/Abstract Syntax Tree/Source/tokenizer/token.cpp
#include "token.h"
#include <stddef.h>


Token::Token(Kind k, const std::string & t, const Range & r) : kind(k), text(t), range(r)
{
	next = NULL;
}

Token::~Token()
{
	if (next)
		delete next;
}



Token::operator std::string ()
{
	return text;
}

File: language/php/lib/Token.php
<?php

class Token
{
	public $type;
	public $range;
	public $text;
	
	public function is($type, $text = null)
	{
		return ($this->type == $type && (!$text || $this->text == $text));
	}
	
	public function __toString()
	{
		return "{$this->type} '{$this->text}'";
	}
}
